### PR TITLE
[release_dashboard] Initial layout and theme

### LINKED ISSUE #90905: Flutter Conductor Desktop UI
## Use case
The current CLI conductor is not very user-friendly. A new desktop app will be made with a simpler UI.

## Proposal

[Project Design Doc](https://docs.google.com/document/d/15AwPXNd5FvItAqM0wa2VK0tRrqtRTgM8vR5LQeT2Mag/edit?resourcekey=0-yZ2FAN-wEKwKT-ymdisetA)

### PR DESCRIPTION
**Pull request by Yugue**
*Created the basic structure and theme for the desktop conductor UI*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

Linked Issue: #90905 

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/conductor/ui/lib/main.dart
Comment (christopherfujino) [inline dev/conductor/ui/lib/main.dart:30]: we should handle if this is false. I would actually check if it's *not* a supported platform, and then throw with a helpful message in that situation.

Comment (Yugue) [reply dev/conductor/ui/lib/main.dart:30]: Ok, you mean another unit test to check if the platform is correct?

File: dev/conductor/ui/lib/main.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/main.dart:31]: If we're failing early, this shouldn't be necessary (the default size should be ok)

File: dev/conductor/ui/test/widget_test.dart
Comment (CaseyHillers) [inline dev/conductor/ui/test/widget_test.dart:8]: nit: rename the file main_test.dart

Tests should match to where the relevant code is

File: dev/conductor/ui/test/widget_test.dart
Comment (CaseyHillers) [inline dev/conductor/ui/test/widget_test.dart:14]: nit: general testing -> main app

File: dev/conductor/ui/test/widget_test.dart
Comment (christopherfujino) [inline dev/conductor/ui/test/widget_test.dart:34]: we don't run CI tests on the web, this is not necessary

Comment (CaseyHillers) [inline dev/conductor/ui/test/widget_test.dart:34]: `kIsWeb || Platform.isWindows`

Comment (christopherfujino) [reply dev/conductor/ui/test/widget_test.dart:34]: for more context, test files run on the HOST device, not the TARGET. we DO have web tests, but those are run from either a Linux, Windows or Mac host, and they then compile web target code.

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/widgets/progression.dart:9]: Add a dart doc

https://dart.dev/guides/language/effective-dart/documentation

File: dev/conductor/ui/pubspec.yaml
Comment (christopherfujino) [inline dev/conductor/ui/pubspec.yaml:29]: sorry, I should have mentioned that you should try to avoid adding any pub dependencies. since the flutter/flutter repository is a monorepo, any version constraint in one dart package constrains all of them. this could make upgrading pub dependencies more difficult.

Also, since this repository is not owned by a Google team, we may need a security review. Are you only using this to set the initial windows size?

Comment (Yugue) [reply dev/conductor/ui/pubspec.yaml:29]: yes, I am only using it to set the window size. Do you know a way without using this package?

Comment (christopherfujino) [reply dev/conductor/ui/pubspec.yaml:29]: Not in Dart, I suspect you would need to use native code to do this. If we just want to set the initial window size, maybe you could just add it to the linux and macos platform code?

Comment (christopherfujino) [reply dev/conductor/ui/pubspec.yaml:29]: Also note Casey's comment--can we not just rely on the default size?

File: dev/conductor/ui/lib/main.dart
Comment (christopherfujino) [reply dev/conductor/ui/lib/main.dart:30]: No, I mean right here. If we ever reach this point on web, the app will just close, and the user won't know why.

File: dev/conductor/ui/pubspec.yaml
Comment (Yugue) [reply dev/conductor/ui/pubspec.yaml:29]: we could, but the user could minimize the window to a really small size, and causing some overflow errors.

Comment (Yugue) [reply dev/conductor/ui/pubspec.yaml:29]: https://user-images.githubusercontent.com/20194490/135176696-d08ccf34-87f5-446c-9f58-7b35f9aeeb70.mov

Comment (CaseyHillers) [reply dev/conductor/ui/pubspec.yaml:29]: Let's ignore this for now, and revisit if we get feedback on it

File: dev/conductor/ui/lib/main.dart
Comment (CaseyHillers) [reply dev/conductor/ui/lib/main.dart:30]: You could do something like this at the start of main to exit
```
if (!(io.Platform.isMacOS || io.Platform.isLinux)) {
  throw Exception('The release dashboard is only support on MacOS and Linux Desktop');
}
```

You can rely on the conductor package failing when they try to build web (dart:io won't be available and will be a compilation error)

Comment (Yugue) [reply dev/conductor/ui/lib/main.dart:29]: I made an error widget for this.

File: dev/conductor/ui/lib/main.dart
Comment (christopherfujino) [inline dev/conductor/ui/lib/main.dart:66]: Is this font already bundled with Flutter?

File: dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj
Comment (christopherfujino) [inline dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj:24]: i'm guessing we don't need all of these diffs from the ui/macos directory, since we're no longer using the pub package?

**Comment by christopherfujino** [comment]
On Linux the tabs are not rendered properly:

Screenshot from 2021-09-28 16-47-51

Is this what you see on mac? I don't think this is a blocker necessarily to this PR, but certainly something we should come back and clean up.

**Comment by Yugue** [comment]
> On Linux the tabs are not rendered properly:
> 
> Screenshot from 2021-09-28 16-47-51
> 
> Is this what you see on mac? I don't think this is a blocker necessarily to this PR, but certainly something we should come back and clean up (ideally this information would be presented prettier anyway)

I don't see the `HT` on Mac. It doesn't matter for now. I will eventually make a UI to present the state nicely.

File: dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj
Comment (Yugue) [reply dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj:24]: There are some build errors I am seeing now. They should be fixed now.

File: dev/conductor/ui/lib/main.dart
Comment (Yugue) [reply dev/conductor/ui/lib/main.dart:66]: Oh you are right, I need to import it. I removed this font. The default font looks good :)

Comment (christopherfujino) [reply dev/conductor/ui/lib/main.dart]: > > On Linux the tabs are not rendered properly:
> > Screenshot from 2021-09-28 16-47-51
> > Is this what you see on mac? I don't think this is a blocker necessarily to this PR, but certainly something we should come back and clean up (ideally this information would be presented prettier anyway)
> 
> I don't see the `HT` on Mac. It doesn't matter for now. I will eventually make a UI to present the state nicely.

Removing the "Montserrat" font fixed this :)

**Comment by christopherfujino** [comment]
> > On Linux the tabs are not rendered properly:
> > Screenshot from 2021-09-28 16-47-51
> > Is this what you see on mac? I don't think this is a blocker necessarily to this PR, but certainly something we should come back and clean up (ideally this information would be presented prettier anyway)
> 
> I don't see the `HT` on Mac. It doesn't matter for now. I will eventually make a UI to present the state nicely.

Removing the "Montserrat" font fixed this :)

**Comment by Yugue** [comment]
@christopherfujino Removing the new MacOS build results in this error. I am reverting it back.
image

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/widgets/progression.dart:23]: This is vague, and would help to have an outline of the release here

```
/// 1. Create candidate branch
/// 2. ...
/// 3. ...
````

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/widgets/progression.dart:12]: Requiring releaseState and making it nullable are counter to each other, choose one

File: dev/conductor/ui/lib/main.dart
Comment (Yugue) [reply dev/conductor/ui/lib/main.dart:54]: Is it ok if we leave the theming as it is for now. I just made some preliminary theming to help me visualize, they will be subject to change.

File: dev/conductor/ui/lib/main.dart
Comment (Yugue) [reply dev/conductor/ui/lib/main.dart:56]: Do we need to have both dark and light themes? Seems a bit extra for this desktop conductor.

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (Yugue) [reply dev/conductor/ui/lib/widgets/progression.dart:12]: I am a little confused. There is a possibility that `releaseState` is null. But we do require it to be passed to the widget no matter what. If we only make `releaseState` nullable and not required, then there could be a possibility that a non-null `releaseState` is not passed to the widget.

**Comment by christopherfujino** [comment]
> @christopherfujino Removing the new MacOS build results in this error. I am reverting it back. image

To fix this you need to `flutter clean`. There are cached build files from a previous `flutter run` that are depending on the unused macos files.

**Review by christopherfujino — CHANGES_REQUESTED** [review]
Per https://github.com/flutter/flutter/pull/90887#issuecomment-930381479, you should remove the Podfile and updated xcode config files.

File: dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj
Comment (christopherfujino) [inline dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj:57]: was this line supposed to be removed?

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/widgets/progression.dart:25]: ... should mirror what the truth is from conductor

Since they won't be implemented this PR, this is a good time to file GitHub issues against yourself and add them as TODOs in the code

File: dev/conductor/ui/lib/main.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/main.dart:29]: This should be the first check in main()

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (CaseyHillers) [reply dev/conductor/ui/lib/widgets/progression.dart:9]: Dart docs on all public members, even if they do something like

```
/// ...
///
/// See [MainProgressionState] for available data.
```

File: dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj
Comment (Yugue) [reply dev/conductor/ui/macos/Runner.xcodeproj/project.pbxproj]: I added it back

File: dev/conductor/ui/lib/widgets/progression.dart
Comment (Yugue) [reply dev/conductor/ui/lib/widgets/progression.dart:12]: Done

File: dev/conductor/ui/lib/main.dart
Comment (CaseyHillers) [inline dev/conductor/ui/lib/main.dart:2]: In the PR message, link a GitHub issue.

You should have access to https://github.com/flutter/flutter/issues/new and can assign the issue to yourself

**Review by CaseyHillers — COMMENTED** [review]
This mostly LGTM, just one last comment from my side on the PR message

**Review by christopherfujino — APPROVED** [review]
LGTM, pending casey's comment regarding linking an issue

File: dev/conductor/ui/lib/main.dart
Comment (Yugue) [reply dev/conductor/ui/lib/main.dart:2]: Hi Casey, are you talking about the [theming](https://github.com/flutter/flutter/issues/90982) and [progression doc issues](https://github.com/flutter/flutter/issues/90981) I created? They are not being addressed in this PR. Or you are referring to this [issue](https://github.com/flutter/flutter/issues/90905)? I linked the latter on the right side panel. I can add it to the PR message as well if that's what you mean.

Comment (CaseyHillers) [reply dev/conductor/ui/lib/main.dart:2]: https://github.com/flutter/flutter/issues/90905 would work well as the linked issue in the PR body

Part of the contributing docs ask for an issue as that's a better format for having discussions (such as if something breaks from the PR)

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**
